views/default/js/file_tools/tree.js

```diff
@@ -0,0 +1,264 @@
+
+					return true;
+				},
+				onmove: function(node, ref_node, type, tree_obj, rb) {
+					var refresh = false;
+					var parent_node = tree_obj.parent(node);
+
+					var folder_guid = $(node).find('a:first').attr('href').substr(1);
+					var parent_guid = $(parent_node).find('a:first').attr('href').substr(1);
+
+					var order = [];
+					$(parent_node).find('>ul > li > a').each(function(k, v) {
+						var guid = $(v).attr('href').substr(1);
+						order.push(guid);
+					});
+
+					if ((parent_guid == window.location.hash.substr(1)) || (file_tools_old_parent_guid == window.location.hash.substr(1))) {
+						if (parent_guid == window.location.hash.substr(1)) {
+							refresh = parent_guid;
+						} else if (file_tools_old_parent_guid == window.location.hash.substr(1)) {
+							refresh = file_tools_old_parent_guid;
+						}
+
+						$("#file_tools_list_files_container .elgg-ajax-loader").show();
+					}
+
+					elgg.action("file_tools/folder/reorder", {
+						data: {
+							folder_guid: folder_guid,
+							parent_guid: parent_guid,
+							order: order
+						},
+						success: function() {
+							if (refresh !== false) {
+								file_tools.load_folder(refresh);
+							}
+						}
+					});
+				}
+			}
+		}).find("a").droppable({
+			accept: "#file_tools_list_files .file-tools-file",
+			hoverClass: "file-tools-tree-droppable-hover",
+			tolerance: "pointer",
+			drop: function(event, ui) {
+				droppable = $(this);
+				draggable = ui.draggable;
+
+				drop_id = droppable.attr("href").substring(1);
+				drag_id = draggable.parent().attr("id").split("-").pop();
+
+				move_file(drag_id, drop_id, draggable);
+			}
+		});
+	};
+
+	/**
+	 *
+	 */
+	function file_tools_get_selected_tree_folder_id(){
+		var result = 0;
+
+		tree = jQuery.tree.reference($("#file_tools_list_tree"));
+		result = file_tools_tree_folder_id(tree.selected);
+		return result;
+	}
+
+	/**
+	 *
+	 * @param {Object} link
+	 */
+	function file_tools_remove_folder_files(link) {
+		if (confirm(elgg.echo("file_tools:folder:delete:confirm_files"))) {
+			var cur_href = $(link).attr("href");
+			$(link).attr("href", cur_href + "&files=yes");
+		}
+		return true;
+	}
+
+	/**
+	 *
+	 * @param {Object} node
+	 * @param {Object} parent
+	 */
+	function file_tools_tree_folder_id(node, parent) {
+		if (parent == true) {
+			var find = "a:first";
+		} else {
+			var find = "a";
+		}
+
+		var element_id = node.find(find).attr("id");
+		return element_id.substring(24, element_id.length);
+	}
+
+	/**
+	 *
+	 * @param {Object} folder_guid
+	 * @param {Object} tree
+	 */
+	function file_tools_select_node(folder_guid, tree) {
+		tree = jQuery.tree.reference($("#file_tools_list_tree"));
+
+		tree.select_branch($("#file_tools_tree_element_" + folder_guid));
+		tree.open_branch($("#file_tools_tree_element_" + folder_guid));
+	}
+
+	/**
+	 *
+	 */
+	$(function() {
+		// TODO(juho) get rid of the PHP
+		//<?php if (file_tools_use_folder_structure()) { ?>
```

Comment:
Another approach could be to include the javascript here always and check the setting as before but from javascript instead of PHP.

Not sure which is more efficient:
 - Loading it always regardless if its needed or not
 - Loading it on demand, which causes one more request to the server